tests/test_lightningd.py

```diff
@@ -873,6 +873,33 @@ def test_pay(self):
         assert len(l1.rpc.listpayments(inv)['payments']) == 1
         assert l1.rpc.listpayments(inv)['payments'][0]['payment_preimage'] == preimage['preimage']
 
+    def test_pay_optional_args(self):
+        l1,l2 = self.connect()
+
+        chanid = self.fund_channel(l1, l2, 10**6)
+ 
+        # Wait for route propagation.
+        self.wait_for_routes(l1, [chanid])
+
+        inv1 = l2.rpc.invoice(123000, 'test_pay', 'description')['bolt11']
+        l1.rpc.pay(inv1, description='00000')
```

Comment:
`description` should match the description in the invoice.

Reply:
Should this rather be verified on the server side? If yes, than ideally it should return an error if description does not match and we then will test for both case.
If the description will be removed from pay command, then I would rather keep this test as is and updated to cover an existing valid case and updated it once we remove description.
I can also take care of description in separate commits.

Reply:
`description` is checked against description hash if the invoice uses description hash, otherwise `description` is ignored.  When generating an invoice, currently the c-lightning checks if the description is more than 640 bytes, to determine if it uses description hash, or embeds description.

Thus to properly check the use of `description`, you should make a 641-byte description in the `invoice` command, and provide the same invoice to `pay` command.